app/commands/user_track/generate_exercise_status_mapping.rb

```diff
@@ -16,7 +16,7 @@ def mapping
       mapping = Hash.new { |k, v| k[v] = Set.new }
 
       Exercise::TaughtConcept.joins(:exercise, :concept).
-        where('exercises.track_id': track.id).
+        where('exercises.id': user_track.concept_exercises).
```

Comment:
Can you check the logs to ensure this does an inner query pls (rather than passing all the ids in). 

(You can see the SQL when you run the command in Rails c)

Reply:
Yep, it uses a subquery:

```sql
SELECT `track_concepts`.`slug`, `exercises`.`slug`
FROM `exercise_taught_concepts`
INNER JOIN `exercises` ON `exercises`.`id` = `exercise_taught_concepts`.`exercise_id`
INNER JOIN `track_concepts` ON `track_concepts`.`id` = `exercise_taught_concepts`.`track_concept_id`
WHERE `exercises`.`id` IN (SELECT `exercises`.`id` FROM `exercises` WHERE `exercises`.`type` = 'ConceptExercise' AND `exercises`.`track_id` = 1 AND (`exercises`.`status` IN (2, 1) OR `exercises`.`id` IN (SELECT `solutions`.`exercise_id` FROM `solutions` INNER JOIN `exercises` ON `exercises`.`id` = `solutions`.`exercise_id` WHERE `solutions`.`user_id` = 4 AND `exercises`.`track_id` = 1)))
 ```

Reply:
Thanks!